arch/riscv/drivers/generic-blkdev.c

```diff
@@ -72,6 +72,7 @@ static void generic_blkdev_process_completions(struct generic_blkdev_port *port)
 		BUG_ON(list_empty(&port->reqbuf[tag]));
 		breq = list_entry(port->reqbuf[tag].prev,
 				struct generic_blkdev_request, list);
+		mb();
```

Comment:
if this does anything useful it's an indication of a much more insidious problem